sda-commons-server-s3/src/main/java/org/sdase/commons/server/s3/health/S3HealthCheck.java

```diff
@@ -0,0 +1,30 @@
+package org.sdase.commons.server.s3.health;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.codahale.metrics.health.HealthCheck;
+import com.codahale.metrics.health.annotation.Async;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Async(period = 30, scheduleType = Async.ScheduleType.FIXED_DELAY)
+public class S3HealthCheck extends HealthCheck {
+
+  private static final Logger LOG = LoggerFactory.getLogger(S3HealthCheck.class);
+
+  private final AmazonS3 s3Client;
+
+  public S3HealthCheck(AmazonS3 s3Client) {
+    this.s3Client = s3Client;
+  }
+
+  @Override
+  protected Result check() {
+    try {
+      s3Client.listBuckets();
```

Comment:
Not sure if this can be a problem with regards to privileges. See: https://aws.amazon.com/de/premiumsupport/knowledge-center/s3-access-denied-listobjects-sync/

Is there any easy alternative that only access one single bucket? Do we need a new config option "bucket for healthcheck" in that case?

Reply:
I actually wanted to avoid specifying a bucket with this, but if that's the case, I guess I can't avoid it.

Reply:
I'm not really sure if it's really a security issue. I'm just cautious because of the latest issues we had with Kafka ACLs.
What about making "healthCheckBucket" an option? We could either use `listBuckets` or make a query to `healthCheckBucket` if someone configured it.